Add ability to manually trigger completion menu

jeremija opened the pull request:
Following the discussion in #1830, a part of `ale#completion#GetCompletions` was extracted into a new function: `ale#completion#AlwaysGetCompletions`. A new `:ALEComplete` command was added, along with a plug mapping: `<Plug>(ale_complete)`. Auto-completion can be triggered manually whether or not `g:ale_completion_enabled` is set.

Sample bindings:

```vim
inoremap <silent> <C-Space> <C-\><C-O>:GetCompletions<CR>
" or
imap <C-Space> <Plug>(ale_complete) 
```

w0rp commented on doc/ale.txt:
```diff
@@ -2208,6 +2208,17 @@ ALE will use to search for Python executables.
+  Manually trigger LSP autocomplete and show the menu. Works only when called
+  from insert mode. >
+
+    inoremap <silent> <C-Space> <C-\><C-O>:GetCompletions<CR>
```
```suggestion
    inoremap <silent> <C-Space> <C-\><C-O>:ALEComplete<CR>
```

I think you need to update the command here.

jeremija replied:
Thanks, that should be fixed now!

jeremija commented:
Hi @w0rp, I updated the docs and added the test. In the test I noticed that I cannot override the `ale#completion#AlwaysGetCompletions()` function without it being called first, because it is an autoloaded function. Do you know of a better way to do this, so that I can remove the initial call in the test?

w0rp commented:
If you rebase your code, that test on Windows will pass.

You can replace the function in Vader tests. Look through existing tests for completion. You can define the function in `Before` or in an `Execute` block, and then reload the original function again with `runtime autoload/ale/completion.vim` in `After`.

jeremija commented:
I guess I'm confused why I get this error in Vim 8.1:
```
Vim(function):E746: Function name does not match script file name: ale#completion#AlwaysGetCompletions
```

for this simple test case:

```
Before:
  let g:get_completions_called = 0
  function! ale#completion#AlwaysGetCompletions() abort
    let g:get_completions_called = 1
  endfunction

Execute(ale#completion#AlwaysGetCompletions should be called when ALEComplete is executed):
  AssertEqual 0, g:get_completions_called
  ALEComplete
  AssertEqual 1, g:get_completions_called
```

But the following works:

```
Before:
  function! MockAlwaysGetCompletions() abort
    let g:get_completions_called = 0

    function! ale#completion#AlwaysGetCompletions() abort
      let g:get_completions_called = 1
    endfunction
  endfunction

  call MockAlwaysGetCompletions()

Execute(ale#completion#AlwaysGetCompletions should be called when ALEComplete is executed):
  AssertEqual 0, g:get_completions_called
  ALEComplete
  AssertEqual 1, g:get_completions_called

```

w0rp commented:
You'll need to add a `runtime` line before you define the function, probably, or try moving it to the `Execute` block. Have a look at existing test files. They do similar things and the tests pass.

jeremija commented:
Thanks, I think I've already tried that and it didn't work. Did you get a chance to take a look at the latest commit? I did something similar to what I wrote above and the tests pass.

w0rp commented:
Cheers! :beers:
